js/src/common/components/ModalManager.js

```diff
@@ -1,40 +1,72 @@
 import Component from '../Component';
+import { createFocusTrap } from '../utils/focusTrap';
+import { tabbable } from 'tabbable';
+
+window.tabbable = tabbable;
 
 /**
  * The `ModalManager` component manages a modal dialog. Only one modal dialog
  * can be shown at once; loading a new component into the ModalManager will
  * overwrite the previous one.
  */
 export default class ModalManager extends Component {
+  /**
+   * @type {import('../utils/focusTrap').FocusTrap | undefined}
+   */
+  focusTrap;
+
+  modalShown = false;
+
   view() {
     const modal = this.attrs.state.modal;
 
     return (
       <div className="ModalManager modal fade">
-        {modal
-          ? modal.componentClass.component({
-              ...modal.attrs,
-              animateShow: this.animateShow.bind(this),
-              animateHide: this.animateHide.bind(this),
-              state: this.attrs.state,
-            })
-          : ''}
+        {!!modal &&
+          modal.componentClass.component({
+            ...modal.attrs,
+            animateShow: this.animateShow.bind(this),
+            animateHide: this.animateHide.bind(this),
+            state: this.attrs.state,
+          })}
       </div>
     );
   }
 
+  /**
+   * @param {import('mithril').VnodeDOM<{}, this>} vnode
+   */
   oncreate(vnode) {
     super.oncreate(vnode);
 
     // Ensure the modal state is notified about a closed modal, even when the
     // DOM-based Bootstrap JavaScript code triggered the closing of the modal,
     // e.g. via ESC key or a click on the modal backdrop.
     this.$().on('hidden.bs.modal', this.attrs.state.close.bind(this.attrs.state));
+
+    console.log(this.element);
+
+    this.focusTrap = createFocusTrap(this.element);
+  }
+
+  onupdate(vnode) {
+    super.onupdate(vnode);
+
+    requestAnimationFrame(() => {
+      try {
+        if (this.modalShown) this.focusTrap.activate?.();
+        else this.focusTrap.deactivate?.();
+      } catch {
+        // We can expect errors to occur here due to the nature of mithril rendering
```

Comment:
Could we catch a more specific kind of exception?

Reply:
I thought about that, but thought the added code needed to check if the exception was the one we'd expect wasn't worth it.

JS doesn't have error classes like PHP and other languages, meaning the only way to catch this specific error would be comparing the error message against a constant.

![image](https://user-images.githubusercontent.com/7406822/129459727-a9a2da0f-6b0a-476a-a2da-2e944a66306f.png)

Reply:
> JS doesn't have error classes like PHP and other languages

Never thought I'd see you admit an upside of PHP :wink:

Reply:
Never ever mention it again. 😄

Reply:
Hm I don't like the idea of catching all errors and not just the one we're trying to avoid spamming...